### PR TITLE
refactor dolphyn for less memory and better performance

### PR DESCRIPTION
**Pull request by bennettm**
This branch is for rewriting AnnualSum values for some of HSC output writing files.  The more efficient way utilizing broadcast replaces for loops so to reduce memory usage and perform in less time.

### CODE REVIEW

**Comment by dharik13** [comment]
@Betristor  let us know when you think you might be able to review this code and share your comments. It would be great to get your feedback this by end of the week.


Separately, I would also suggest testing whether you get the same results with this code vs. develop branch using your autotest module.

File: src/HSC/write_outputs/write_h2_nse.jl [hunk 1]
```diff
@@ -31,11 +31,12 @@ function write_h2_nse(path::AbstractString, sep::AbstractString, inputs::Dict, s
         dfTemp = DataFrame(Segment=zeros(H2_SEG), Zone=zeros(H2_SEG), AnnualSum = Array{Union{Missing,Float32}}(undef, H2_SEG))
         dfTemp[!,:Segment] = (1:H2_SEG)
         dfTemp[!,:Zone] = fill(z,(H2_SEG))
+
+            h2nse = value.(EP[:vH2NSE])[:, :, z]
```
Comment (Betristor) [inline src/HSC/write_outputs/write_h2_nse.jl:35]: I think this line could be moved out of the code block and then index could be dropped. The value process is time and resource consuming.

**Comment by Betristor** [comment]
@bennettm @dharik13 This auto test check could be automatically triggered after branch AutoTest is merged. The workflow should be in place only after changes are applied to main and develop branches.

**Comment by Betristor** [comment]
@bennettm @dharik13 I checked out necessary changes from AutoTest branch to trigger test for this PR.

**Comment by RuaridhMacd** [comment]
- [ ] @Betristor Roll-back AutoTest commits so we can push just the output-writing updates

**Comment by Betristor** [comment]
@RuaridhMacd Sorry that I didn't have time to withdraw those changes. But yes only four sequential commits are applied and with ```git revert``` command the commits could be reverted. Much thanks @RuaridhMacd.

**Comment by RuaridhMacd** [comment]
No problem @Betristor 
I wanted to try some changes myself so just went ahead

**Review by RuaridhMacd — APPROVED** [review]
This all looks great and tests correctly

**Pull request merged**